web-ui/src/app/components/latest-blocks/latest-blocks.component.ts

```diff
@@ -72,28 +71,16 @@ export class LatestBlocksComponent implements OnInit, OnDestroy {
   }
 
   private onBlockRetrieved(response: Block[]) {
-    this.countNewestBlocks(response);
-    this.latestBlock = response[0];
+    this.newBlocks = this.getNewBlocks(response);
     this.blocks = response;
   }
 
-  private countNewestBlocks(newBlocks: Block[]) {
+  private getNewBlocks(newBlocks: Block[]): Block[] {
     if (!this.blocks) {
-      return;
+      return [];
     }
 
-    const lastestBlockHash: string = this.latestBlock.hash;
-    let totalNewBlocks = 0;
-
-    for (let i = 0; i < newBlocks.length; i++) {
-      if (newBlocks[i].hash === lastestBlockHash) {
-        break;
-      }
-
-      totalNewBlocks++;
-    }
-
-    this.totalNewestBlocks = totalNewBlocks;
+    return newBlocks.filter(newBlock => newBlock.height > this.blocks[0].height);
```

Comment:
you are assuming that the blocks are in certain order, while today that's true, we don't know if it would be tomorrow. you can easily get the current latest block height and compare against that.